solr/core/src/java/org/apache/solr/cluster/placement/PlacementPluginFactory.java

```diff
@@ -18,14 +18,22 @@
 package org.apache.solr.cluster.placement;
 
 /**
- * Factory implemented by client code and configured in {@code solr.xml} allowing the creation of instances of
+ * Factory implemented by client code and configured in container plugins allowing the creation of instances of
  * {@link PlacementPlugin} to be used for replica placement computation.
+ * <p>Note: configurable factory implementations should also implement
+ * {@link org.apache.solr.api.ConfigurablePlugin} with the appropriate configuration
+ * bean type.</p>
  */
 public interface PlacementPluginFactory {
```

Comment:
Shouldn't this interface extend `ConfigurablePlugin<T extends MapWriter>` so that concrete config classes such as `AffinityPlacementFactory` do not have to implement multiple interfaces?

Reply:
That would force every placement plugin to be configurable though I guess...